index.js

```diff
@@ -73,24 +73,27 @@
           }
           eventHandler(evt);
         }
-      }(ReactDOM.findDOMNode(this), this.handleClickOutside));
+      }(ReactDOM.findDOMNode(this), this.handleOutsideEvent));
 
       var pos = registeredComponents.length;
       registeredComponents.push(this);
       handlers[pos] = fn;
 
-      // If there is a truthy disableOnClickOutside property for this
-      // component, don't immediately start listening for outside events.
-      if (!this.props.disableOnClickOutside) {
-        this.enableOnClickOutside();
+      // Only start immediately listening for outside events if there is no
+      // truthy disableOnOutsideEvent prop, and a valid listenToOutsideEvent
+      // prop exists.
+      var evtTypeProp = this.props.listenToOutsideEvent;
+      if (!this.props.disableOnOutsideEvent &&
+          (typeof evtTypeProp === 'string' || evtTypeProp instanceof Array)) {
```

Comment:
I'm tempted to say "no need to check, the onus is on the user here to not pass in things that obviously won't work". Either that, or tell the user what they did wrong in an `else`. The conditional as is will allow for silent failure.